### ISSUE TITLE
Optimizing the Latent Space of Generative Networks

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#120]
*ICML 2018*

# TL;DR

- Train generator and latent code simultaneously using ***reconstruction loss*** *(Laplacian Pyramid + L2)*
- Learn an additional probabilistic model for sampling Z (learned latent) from prior distribution (like Gaussian)
- Properties like sharp edge, latent space interpolation and generative power are not children of adversarial learning, but rather the structure of convolutional neural nets.

# Motivation

- Disentangle the effect of network architecture (CNN) and adversarial training (discriminator) in image generation

# Previous Work

- GANs tend to produce realistic images (sharp edges) while causing much trouble in training.
- Successful GANs = ConvNet + Adversarial Learning

# Method

- Learn generator parameters as well as latent vectors
![image](https://user-images.githubusercontent.com/46308407/131439701-7d43871d-069f-4b01-88df-1a16722351c8.png)

- For reconstruction loss, use combination of Laplacian loss and L2 loss.
![image](https://user-images.githubusercontent.com/46308407/131439989-298fabc2-c5aa-4e9b-802e-7e43111220b4.png)

- Note that Laplacian pyramid incorporates multi-level features of an image, hence enabling sharp image output like GANs.
- *Also note that GANs use CNN(=discriminator) as loss function, allowing for shape information in lower level feature maps.*

- Generator architecture is that of DCGAN.

- So, for a set of N image samples, N latent vectors are learned.
- Fit a single full covariance Gaussian distribution to the learned Zs, for generative purpose.

- Project z to the unit sphere after each update, and optimize using SGD.

# Experiments

![image](https://user-images.githubusercontent.com/46308407/131440099-82527bfd-a96a-436c-abc0-f349576aff8b.png)
![image](https://user-images.githubusercontent.com/46308407/131440114-45a8e9db-dd17-41ef-a35a-c40b9cf0b860.png)
![image](https://user-images.githubusercontent.com/46308407/131440136-a566ec76-4f0e-4e43-98ee-117b83de3177.png)

- Note that VAEs and GLO fails on LSUN bedroom. GAN's success is attributable to its ***feature of dropping modes***.
- When a dataset gets complex, model is unable to learn the full distribution of data. As a result, VAEs and GLOs degrade output quality since they cannot withstand mode dropping, while GANs retain sample quality at the expense of diversity.

![image](https://user-images.githubusercontent.com/46308407/131440450-d54b5153-54de-4783-aeb0-01e817fc528c.png)

- Note that DCGAN fails to reconstruct many samples in LSUN bedroom. It can be said that GANs were not trained to reconstruct, but it also indicates that GANs selectively learn effective modes of a complex dataset while ignoring the rest. 
- In contrast, though output quality maybe blurry in general, models like VAE and GLOs do not ***totally ignore*** certain part of the distribution.

# Comments

- Interesting question: is beneficial properties of GANs the result of adversarial learning or of convolutional network?
- Their analysis on mode collapse and distribution coverage is not only interesting but very persuasive.
- GANs do produce realistic outputs, but given restricted model capacity, it selectively learns certain portion of the true distribution, hence sacrificing diversity.
- Somewhat similar to ***Deep Image Prior***. Surprising how a simple reconstruction loss can be as powerful as shown here.
